src/main/java/com/williamfiset/algorithms/datastructures/binarysearchtree/SplayTree.java

```diff
@@ -283,14 +283,14 @@ public static void main(String[] args) {
     SplayTree<Integer> splayTree = new SplayTree<>();
     Scanner sc = new Scanner(System.in);
     int[] data = {2, 29, 26, -1, 10, 0, 2, 11};
-
+    int c = 0;
     for (int i : data) {
       splayTree.insert(i);
     }
 
-    while (true) {
-      System.out.println("1. Insert 2. Delete 3. Search 4.FindMin 5.FindMax 6. PrintTree");
-      int c = sc.nextInt();
+    while (c!=7) {
```

Comment:
you don't need c!=7 since you're breaking below, so you can keep c scoped within the loop

Reply:
No, you can't. Since while (true) never terminates, you need a variable for the termination condition, that can't be within in the loop

Reply:
You can add a loop tag:

```java
outerloop:
while (true) {

   ...

   case 7:
      break outerloop;
}
```